packages/gateway/src/features/search/root-resolvers.ts

```diff
@@ -94,31 +94,47 @@ export const resolvers: GQLResolver = {
           SCOPES.SEARCH_MARRIAGE,
           SCOPES.SEARCH_BIRTH_MY_JURISDICTION,
           SCOPES.SEARCH_DEATH_MY_JURISDICTION,
-          SCOPES.SEARCH_MARRIAGE_MY_JURISDICTION
+          SCOPES.SEARCH_MARRIAGE_MY_JURISDICTION,
+          SCOPES.RECORDSEARCH
         ])
       )
         return {
           totalItems: 0,
           results: []
         }
 
+      // TODO: refactor this concept to avoid calling the dataSource.usersAPI
       const userIdentifier = getTokenPayload(authHeader.Authorization).sub
-      const user = await dataSources.usersAPI.getUserById(userIdentifier!)
-      const office = await dataSources.locationsAPI.getLocation(
-        user.primaryOfficeId
-      )
-      const officeLocationId =
-        office.partOf?.reference &&
-        resourceIdentifierToUUID(office.partOf.reference)
+      let user
+      let system
+
+      try {
+        user = await dataSources.usersAPI.getUserById(userIdentifier!)
+      } catch (error) {
+        //user not found, attempting to find the system
+        system = await getSystem({ systemId: userIdentifier }, authHeader)
 
-      if (!officeLocationId) {
+        if (!system) {
+          throw new Error('API client system not found')
+        }
+      }
+
+      const office = user
+        ? await dataSources.locationsAPI.getLocation(user.primaryOfficeId)
+        : undefined
+      const officeLocationId = office
+        ? office.partOf?.reference &&
+          resourceIdentifierToUUID(office.partOf.reference)
+        : undefined
+
+      if (user && !officeLocationId) {
         throw new Error('User office not found')
       }
 
       const transformedSearchParams = transformSearchParams(
         getTokenPayload(authHeader.Authorization).scope,
         advancedSearchParameters,
-        officeLocationId
+        ''
```

Comment:
@bvenceslas only if it's the record search client should we send an empty string as the office location but with this change it will always be sending so